### PR TITLE
[#228] | Added toast if a guest tries to vote on a snippet

### PR DESCRIPTION
**Pull request by TyrinH**
---
title: Issue # | [#228 ](https://github.com/webdevcody/code-racer/issues/228)
---

Discord Username: @industrializeddark

## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature
- [X] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description
Displayed toast for if a guest tries to vote on a snippet after completing it, this will suggest the user to sign in.

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: src/app/result/voting.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { toast } from "@/components/ui/use-toast";
 import { catchError, cn } from "@/lib/utils";
 
 interface VotingProps {
-  userId: User["id"];
+  userId: User["id"] | undefined;
```
Comment (scape76) [inline src/app/result/voting.tsx:17]: you can say userId?: User["id"] for that

File: src/app/result/voting.tsx [hunk 2]
```diff
@@ -33,26 +33,34 @@ export function Voting({ userId, snippetId, usersVote }: VotingProps) {
...
-                    userId,
-                    snippetId,
-                  });
-                  toast({
-                    title: "Success.",
-                    description:
-                      "Thanks for your feedback! We will consider it.",
-                    variant: "default",
-                  });
+              if (userId) {
```
Comment (scape76) [inline src/app/result/voting.tsx:36]: I would personally say
if (!userId) return toast(...)

**Comment by scape76** [comment]
Well done

**Comment by TyrinH** [comment]
> Well done

Thank you for your help with this PR, it was my first one sorry for all questions I asked. Do you want me to fix the areas you left feedback on and open another PR again?

**Comment by scape76** [comment]
> > Well done
> 
> Thank you for your help with this PR, it was my first one sorry for all questions I asked. Do you want me to fix the areas you left feedback on and open another PR again?

you can make a quick new pr if you dont mind

**Comment by TyrinH** [comment]
> > > Well done
> > 
> > 
> > Thank you for your help with this PR, it was my first one sorry for all questions I asked. Do you want me to fix the areas you left feedback on and open another PR again?
> 
> you can make a quick new pr if you dont mind

@scape76 Here's the PR with those fixes: [PR #289](https://github.com/webdevcody/code-racer/pull/289)

**Pull request merged**